File: Clases_Instancias/4/modificar_promedio.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Estudiante {
public:
    string nombre;
    int edad;
    float promedio;

    Estudiante(string n, int e, float p) : nombre(n), edad(e), promedio(p) {}

    void mostrarInfo() {
        cout << "Nombre: " << nombre << endl;
        cout << "Edad: " << edad << endl;
        cout << "Promedio: " << promedio << endl;
    }

    void setPromedio(float nuevoPromedio) {
        promedio = nuevoPromedio;
    }
};

int main() {
    // Definir un vector para almacenar las instancias
    vector<Estudiante> listaEstudiantes;

    // Crear 3 instancias y agregarlas al vector
    listaEstudiantes.push_back(Estudiante("Oscar", 20, 4.5));
    listaEstudiantes.push_back(Estudiante("Maria", 19, 4.8));
    listaEstudiantes.push_back(Estudiante("Juan", 21, 3.9));

    cout << "Informacion inicial:" << endl;
    for (int i = 0; i < listaEstudiantes.size(); i++) {
        listaEstudiantes[i].mostrarInfo();
        cout << "--------------------" << endl;
    }

    // Cambiar el promedio del primer estudiante mediante setPromedio
    cout << "\nModificando el promedio de " << listaEstudiantes[0].nombre << " a 4.9..." << endl;
    listaEstudiantes[0].setPromedio(4.9);

    cout << "\nInformacion despues de la modificacion:" << endl;
    for (int i = 0; i < listaEstudiantes.size(); i++) {
        listaEstudiantes[i].mostrarInfo();
        cout << "--------------------" << endl;
    }

    return 0;
}

File: Clases_Instancias/4/modificar_promedio.py
class Estudiante:
    def __init__(self, nombre, edad, promedio):
        self.nombre = nombre
        self.edad = edad
        self.promedio = promedio

    def mostrarInfo(self):
        print(f"Nombre: {self.nombre}")
        print(f"Edad: {self.edad}")
        print(f"Promedio: {self.promedio}")

    def setPromedio(self, nuevo_promedio):
        self.promedio = nuevo_promedio

if __name__ == "__main__":
    # Crear 3 instancias
    estudiante1 = Estudiante("Oscar", 20, 4.5)
    estudiante2 = Estudiante("Maria", 19, 4.8)
    estudiante3 = Estudiante("Juan", 21, 3.9)

    # Almacenarlas en una lista
    lista_estudiantes = [estudiante1, estudiante2, estudiante3]

    print("Información inicial:")
    for est in lista_estudiantes:
        est.mostrarInfo()
        print("-" * 20)

    # Cambiar el promedio del primer estudiante mediante setPromedio
    print(f"\nModificando el promedio de {estudiante1.nombre} a 4.9...")
    estudiante1.setPromedio(4.9)

    print("\nInformación después de la modificación:")
    for est in lista_estudiantes:
        est.mostrarInfo()
        print("-" * 20)
